base/rational.jl

```diff
@@ -14,9 +14,13 @@ Rational(n::T, d::T) where T<:Integer = Rational{T}(n,d)
 Rational(n::Integer, d::Integer) = Rational(promote(n,d)...)
 Rational(n::Integer) = Rational(n,one(n))
 
-function divgcd(x::Integer,y::Integer)
-    g = gcd(x,y)
-    div(x,g), div(y,g)
+function divgcd(x::Integer, y::Integer)
+    if iszero(x) && iszero(y)
+        promote(x, y)
+    else
+        g = gcd(x, y)
+        promote(div(x, g), div(y, g))
+    end
```

Comment:
Would it make more sense to promote to a common type first:
```jl
divgcd(x::Integer, y::Integer) = divgcd(promote(x, y)...)

function divgcd(x::T, y::T) where T<:Integer
    if !iszero(x) || !iszero(y)
        g = gcd(x, y)
        x ÷= g
        y ÷= g
    end
    return x, y
end
```

Reply:
It's possibly intentional that things are done this way, since `div` has special cases for unsigned/signed combinations.

```julia
julia> @which 0x1 ÷ -1
div(x::Unsigned, y::Union{Int128, Int16, Int32, Int64, Int8}) in Base at int.jl:160
```

This behavior might require reviewing for desirability/consistency.